superset-frontend/src/explore/components/useExploreAdditionalActionsMenu/DashboardsSubMenu.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState } from 'react';
+import { css, t, useTheme } from '@superset-ui/core';
+import { AntdInput } from 'src/components';
+import Icons from 'src/components/Icons';
+import { Menu } from 'src/components/Menu';
+import { Link } from 'react-router-dom';
+
+export interface DashboardsSubMenuProps {
+  chartId?: number;
+  dashboards?: { id: number; dashboard_title: string }[];
+}
+
+const WIDTH = 220;
+const HEIGHT = 300;
+const SEARCH_THRESHOLD = 10;
+
+const DashboardsSubMenu = ({
+  chartId,
+  dashboards = [],
+  ...menuProps
+}: DashboardsSubMenuProps) => {
+  const theme = useTheme();
+  const [dashboardSearch, setDashboardSearch] = useState<string>();
+  const [hoveredItem, setHoveredItem] = useState<number | null>();
+  const showSearch = dashboards.length > SEARCH_THRESHOLD;
+  const filteredDashboards = dashboards.filter(
+    dashboard =>
+      !dashboardSearch ||
+      dashboard.dashboard_title
+        .toLowerCase()
+        .includes(dashboardSearch.toLowerCase()),
+  );
+  const noResults = dashboards.length === 0;
+  const noResultsFound = dashboardSearch && filteredDashboards.length === 0;
+  const urlQueryString = chartId ? `?focused_chart=${chartId}` : '';
+  return (
+    <>
+      {showSearch && (
+        <AntdInput
```

Comment:
Should we move this to its own directory and enhance it like we did for all other Antd components? Definitely not a blocker for the PR though

Reply:
What do you mean by "enhance it"?

Reply:
Just move it in its own components directory and wrap it with styled components. As I said, it's just a plus

Reply:
I changed the code to use our styled `Input` from `src/components` 👍🏼